### PR TITLE
feat(EVS): add datasource to query the list of EVS snapshots

### PR DESCRIPTION
**Pull request by zhangting951007**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST="./huaweicloud/services/acceptance/evs" TESTARGS="-run TestAccDatasourceEVSsnapshots_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/evs -v -run TestAccDatasourceEVSsnapshots_basic -timeout 360m -parallel 4
=== RUN   TestAccDatasourceEVSsnapshots_basic
=== PAUSE TestAccDatasourceEVSsnapshots_basic
=== CONT  TestAccDatasourceEVSsnapshots_basic
--- PASS: TestAccDatasourceEVSsnapshots_basic (119.98s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/evs       120.028s
```

### CODE REVIEW

**Comment by zhangting951007** [comment]
```make testacc TEST="./huaweicloud/services/acceptance/evs" TESTARGS="-run TestAccDatasourceEVSsnapshots_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/evs -v -run TestAccDatasourceEVSsnapshots_basic -timeout 360m -parallel 4
=== RUN   TestAccDatasourceEVSsnapshots_basic
=== PAUSE TestAccDatasourceEVSsnapshots_basic
=== CONT  TestAccDatasourceEVSsnapshots_basic
--- PASS: TestAccDatasourceEVSsnapshots_basic (113.87s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/evs       113.914s
```

**Comment by zhangting951007** [comment]
```
make testacc TEST="./huaweicloud/services/acceptance/evs" TESTARGS="-run TestAccDatasourceEVSsnapshots_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/evs -v -run TestAccDatasourceEVSsnapshots_basic -timeout 360m -parallel 4
=== RUN   TestAccDatasourceEVSsnapshots_basic
=== PAUSE TestAccDatasourceEVSsnapshots_basic
=== CONT  TestAccDatasourceEVSsnapshots_basic
--- PASS: TestAccDatasourceEVSsnapshots_basic (123.77s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/evs       123.823s
```

File: docs/data-sources/evs_snapshots.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+* `volume_id` - (Optional, String) The ID of the cloudvolume to which the snapshot belongs.
+
+* `availability_zone` - (Optional, String) The availability zone of the cloudvolume to which the snapshot belongs.
+
+* `snapshot_id` - (Optional, String) Specify the snapshot ID to filter.
+  You can pass in multiple ids to filter the query, for example: id=id1&id=id2&id=id3.
+
+* `dedicated_storage_name` - (Optional, String) The name of the dedicated storage.
+
+* `dedicated_storage_id` - (Optional, String) The id of the dedicated storage.
```
Comment (deer-hang) [inline docs/data-sources/evs_snapshots.md]:
```suggestion
* `dedicated_storage_id` - (Optional, String) The ID of the dedicated storage.
```

File: huaweicloud/services/acceptance/evs/data_source_huaweicloud_evs_snapshots_test.go [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+data "huaweicloud_evs_snapshots" "service_type_filter" {
+  service_type = "evs"
+}
+
+data "huaweicloud_evs_snapshots" "availability_zone_filter" {
+  availability_zone = data.huaweicloud_availability_zones.test.names[0]
+}
+
+data "huaweicloud_evs_snapshots" "dedicated_storage_name_filter" {
+  dedicated_storage_name = resource.huaweicloud_evs_volume.test.dedicated_storage_name
```
Comment (deer-hang) [inline huaweicloud/services/acceptance/evs/data_source_huaweicloud_evs_snapshots_test.go]:
```suggestion
  dedicated_storage_name = huaweicloud_evs_volume.test.dedicated_storage_name
```
check the other parts.

File: huaweicloud/services/acceptance/evs/data_source_huaweicloud_evs_snapshots_test.go [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+    [for v in data.huaweicloud_evs_snapshots.volume_id_filter.snapshots[*].volume_id : v == local.volume_id]
+  )  
+}
+
+data "huaweicloud_evs_snapshots" "snapshot_id_filter" {
+  snapshot_id = data.huaweicloud_evs_snapshots.test.snapshots.0.id
+}
+
+locals {
+	snapshot_id = data.huaweicloud_evs_snapshots.test.snapshots.0.id
```
Comment (deer-hang) [inline huaweicloud/services/acceptance/evs/data_source_huaweicloud_evs_snapshots_test.go]:
```suggestion
  snapshot_id = data.huaweicloud_evs_snapshots.test.snapshots.0.id
```

File: huaweicloud/services/acceptance/evs/data_source_huaweicloud_evs_snapshots_test.go [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+data "huaweicloud_evs_snapshots" "snapshot_id_filter" {
+  snapshot_id = data.huaweicloud_evs_snapshots.test.snapshots.0.id
+}
+
+locals {
+  snapshot_id = data.huaweicloud_evs_snapshots.test.snapshots.0.id
+}
+
+output "snapshot_id_filter_is_useful" {
+  value = length(data.huaweicloud_evs_snapshots.volume_id_filter.snapshots) > 0 && alltrue(
```
Comment (deer-hang) [inline huaweicloud/services/acceptance/evs/data_source_huaweicloud_evs_snapshots_test.go]:
```suggestion
  value = length(data.huaweicloud_evs_snapshots.snapshot_id_filter.snapshots) > 0 && alltrue(
```
Same question down below.

**Comment by zhangting951007** [comment]
```
 make testacc TEST="./huaweicloud/services/acceptance/evs" TESTARGS="-run TestAccDatasourceEVSsnapshots_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/evs -v -run TestAccDatasourceEVSsnapshots_basic -timeout 360m -parallel 4
=== RUN   TestAccDatasourceEVSsnapshots_basic
=== PAUSE TestAccDatasourceEVSsnapshots_basic
=== CONT  TestAccDatasourceEVSsnapshots_basic
--- PASS: TestAccDatasourceEVSsnapshots_basic (113.69s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/evs       113.735s
```

**Pull request merged**